pyanalyze/asynq_checker.py

```diff
@@ -192,6 +199,8 @@ def is_impure_async_fn(value: Value) -> bool:
     """
     if isinstance(value, KnownValue):
         return asynq.is_async_fn(value.val) and not asynq.is_pure_async_fn(value.val)
+    elif isinstance(value, MultiValuedValue):
```

Comment:
You may also want to handle AnnotatedValue and a few others. In my pycroscope fork I did a bit of work to handle these "wrapper" values a bit more sanely: https://github.com/JelleZijlstra/pycroscope/blob/385397ac151199638c65d298997fd8298eb1abb2/pycroscope/value.py#L2360 . It might make sense to adopt that approach in pyanalyze too.